Unpin stactools

TomAugspurger opened the pull request:
Let the application using this library choose the version they want.

Commit:
Unpin stactools

Let the application using this library choose the version they want.

gadomski commented on setup.cfg:
```diff
@@ -27,7 +27,7 @@ package_dir =
 packages = find_namespace:
 install_requires =
-    stactools == 0.2.3
+    stactools
```
```suggestion
    stactools >= 0.2.3
```
Using an older version than the package was developed against could be risky though, yes?

Commit:
Sync template changes

https://github.com/stactools-packages/template/blob/967affc09575072df8d798ff956dda445e05d686/scripts/stactools-version.py

Commit:
fixup

Commit:
fixup

Commit:
unpin black

Commit:
Fixed temporal extent

TomAugspurger commented:
@gadomski all green now. Are you able to merge this?

FYI: https://github.com/stactools-packages/jrc-gsw/pull/18/files#diff-f8ea2be88261fd5306523082385fdb51bb03a6e1c90f3da28e712c1ae15aaa12R326 has an actual change since you last reviewed. Not sure what changed upstream in pystac, but I think the new behavior is correct.
